packages/cozy-harvest-lib/src/datacards/FileDataCard.jsx

```diff
@@ -169,60 +146,11 @@ const FileCard = ({ files, loading, konnector, trigger }) => {
   )
 }
 
-const makeFolderToSaveQueryFromProps = ({ trigger }) => ({
-  query: Q('io.cozy.files')
-    .where({
-      dir_id: trigger.message.folder_to_save,
-      trashed: false
-    })
-    .indexFields(['dir_id', 'cozyMetadata.createdAt'])
-    .sortBy([{ dir_id: 'desc' }, { 'cozyMetadata.createdAt': 'desc' }])
-    .limitBy(5),
-  as: `fileDataCard_io.cozy.files/${trigger.message.folder_to_save}/io.cozy.files`,
-  fetchPolicy: CozyClient.fetchPolicies.olderThan(30 * 1000)
-})
-
-const makeSourceAccountQueryFromProps = ({ accountId }) => ({
-  query: Q('io.cozy.files')
-    .where({
-      'cozyMetadata.sourceAccount': accountId,
-      trashed: false
-    })
-    .indexFields(['cozyMetadata.sourceAccount', 'cozyMetadata.createdAt'])
-    .sortBy([
-      { 'cozyMetadata.sourceAccount': 'desc' },
-      { 'cozyMetadata.createdAt': 'desc' }
-    ])
-    .limitBy(5),
-  as: `fileDataCard_io.cozy.accounts/${accountId}/io.cozy.files`,
-  fetchPolicy: CozyClient.fetchPolicies.olderThan(30 * 1000)
-})
-
-const FileDataCard = ({
-  folderToSaveFiles,
-  sourceAccountFiles,
-  konnector,
-  trigger
-}) => {
-  const { data: files1 } = folderToSaveFiles
-  const { data: files2 } = sourceAccountFiles
-
-  const noFiles =
-    hasQueryBeenLoaded(folderToSaveFiles) &&
-    files1.length === 0 &&
-    hasQueryBeenLoaded(sourceAccountFiles) &&
-    files2.length === 0
-  const isLoading =
-    isQueryLoading(folderToSaveFiles) || isQueryLoading(sourceAccountFiles)
+const FileDataCard = ({ accountId, konnector, trigger }) => {
+  const files = useDataCardFiles(accountId, trigger.id)
+  const isLoading = !files
```

Comment:
This line sounds wrong to me. This is not because we don't have `files` that we're still 'loading'. It creates an indirection between two different things.

Reply:
With my new implementation, null = loading, empty array = loaded but empty. Do you prefer clearer return interface?

Reply:
I prefer to use Cozy-Client mechanism: fetchStatus. How do you know if it's loading or if there is an error somewhere for instance ?

Reply:
- [ ] Will use `type QueryFetchStatus = "failed" | "loading" | "pending" | "loaded";` along with eventual data

Reply:
Don't forget our internal conversation about the `loading` state ;)